client/state/user-devices/reducer.js

```diff
@@ -0,0 +1,11 @@
+/**
+ * Internal dependencies
+ */
+import { createReducer } from 'state/utils';
+import {
+	USER_DEVICES_ADD,
+} from 'state/action-types';
+
+export default createReducer( {}, {
+	[ USER_DEVICES_ADD ]: ( state, { devices } ) => devices
```

Comment:
the problem is that we don't really care about the previous `state`. The API always returns the full list of devices.

as a side note, because of that, maybe `USER_DEVICES_RECEIVE` is a better action name.